src/cls/IPM/Lifecycle/Base.cls

```diff
@@ -640,15 +640,49 @@ Method %Reload(ByRef pParams) As %Status
 			Set ^Sources("MODULE",..Module.Name) = tAbsolutePrefix_tRoot
 		}
 		
-		// Standard resource processing
-		Set tSC = $$$OK
+		// Process resources based on their package depth (higher level first so more granular resource will override)
+		// e.g. <Resource Name="Depth1.PKG"/> will be processed before <Resource Name="Depth1.Depth2.CLS"/>
+		Set nonDocumentResources = []
 		Set tKey = ""
 		For {
 			Set tResource = ..Module.Resources.GetNext(.tKey)
 			Quit:tKey=""
 			If $IsObject(tResource.Processor) {
+				// Strip extension from resource name as that caused improper sorting (i.e. MyPackage.MyClass.CLS gets sorted before MyPackage.PKG)
+				Set tProcessorClassName = tResource.Module.Lifecycle.GetDefaultResourceProcessor(tResource.Name)
```

Comment:
To figure out if the processor extends Default.Document, can simple do the following: 
`If tResource.Processor.%Extends("%IPM.ResourceProcessor.Default.Document")`